plugin/trace/trace.go

```diff
@@ -120,9 +123,28 @@ func (t *trace) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 	span.SetTag(tagType, req.Type())
 	span.SetTag(tagRcode, rcode.ToString(rw.Rcode))
 
+	for key, internalKey := range t.tags {
```

Comment:
Instead of implementing Provider below to handle static values, I think it's simpler to just treat static and dynamic values separately.

That mean:
1) your setup code is simpler, you just need the tags not the tagFetchers
2) this code will just loop through the tags, and it will look and see if the value has { }; if so, it calls ValueFunc; if not, it just writes the value directly
3) you don't need to implement provider

I think that will work, with less code, won't it?